eng/testing/tests.browser.targets

```diff
@@ -96,6 +96,7 @@
     <_AppArgs Condition="'$(WasmTestAppArgs)' != ''">$(_AppArgs) $(WasmTestAppArgs)</_AppArgs>
 
     <WasmXHarnessMonoArgs Condition="'$(XunitShowProgress)' == 'true'">$(WasmXHarnessMonoArgs) --setenv=XHARNESS_LOG_TEST_START=1</WasmXHarnessMonoArgs>
+    <WasmXHarnessMonoArgs Condition="'$(WasmEnableThreads)' == 'true' or '$(WasmEnablePerfTracing)' == 'true' or '$(WasmEnablePerfTracing)' == 'true' or '$(MonoWasmBuildVariant)' == 'multithread'">$(WasmXHarnessMonoArgs) --no-memory-snapshot</WasmXHarnessMonoArgs>
```

Comment:
This chain of 'or's is kind of unwieldy, it might be better to have a synthesized property like
```xml
<WasmSnapshotsAllowed Condition="'$(WasmEnableThreads)' == 'true'">false</WasmSnapshotsAllowed>
...
<WasmSnapshotsAllowed Condition="'$(WasmSnapshotsAllowed)' == ''">true</WasmSnapshotsAllowed>
<WasmXHarnessMonoArgs Condition="'$(WasmSnapshotsAllowed' == 'true'">$(WasmXHarnessMonoArgs) --no-memory-snapshot</WasmXHarnessMonoArgs>
```

Reply:
simplified by removing perftracing flavor